File: Projects/PaymentApplication/app/src/main/java/nuricanozturk/dev/android/payment/LoginActivity.java
package nuricanozturk.dev.android.payment;

import static nuricanozturk.dev.android.payment.global.BundleKeysKt.LOGIN_INFO;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Toast;

import com.karandev.util.data.service.DataServiceException;

import javax.inject.Inject;

import dagger.hilt.android.AndroidEntryPoint;
import nuricanozturk.dev.android.app.data.service.PaymentAppDataService;
import nuricanozturk.dev.android.app.data.service.dto.LoginInfoDTO;
import nuricanozturk.dev.android.payment.databinding.ActivityLoginBinding;
import nuricanozturk.dev.android.payment.viewmodel.LoginActivityListenerViewModel;

@AndroidEntryPoint
public class LoginActivity extends AppCompatActivity
{
    private ActivityLoginBinding m_loginActivityBinding;

    @Inject
    PaymentAppDataService m_service;

    private void init()
    {
        initBinding();
    }

    private void initBinding()
    {
        m_loginActivityBinding = DataBindingUtil.setContentView(this, R.layout.activity_login);
        m_loginActivityBinding.setViewModel(new LoginActivityListenerViewModel(this));
        m_loginActivityBinding.setLoginInfo(new LoginInfoDTO());
    }

    @Override
    protected void onCreate(Bundle savedInstanceState)
    {
        super.onCreate(savedInstanceState);
        init();
    }

    public void loginButtonClicked()
    {
        try
        {
            var loginInfo = m_loginActivityBinding.getLoginInfo();
            if (m_service.checkAndSaveLoginInfo(loginInfo))
            {
                Toast.makeText(this, "Access granted", Toast.LENGTH_LONG).show();
                startActivity(new Intent(this, OperationsActivity.class).putExtra(LOGIN_INFO, loginInfo));
            }
            else Toast.makeText(this, "Access denied!", Toast.LENGTH_LONG).show();

        }
        catch (DataServiceException ex)
        {
            Toast.makeText(this, "Data Problem: " + ex.getMessage(), Toast.LENGTH_LONG).show();
        }
        catch (Throwable ex)
        {
            Toast.makeText(this, "Problem occured: " + ex.getMessage(), Toast.LENGTH_LONG).show();
        }
    }
}

File: MultipleActivity/app/src/main/java/nuricanozturk/dev/android/multipleactivity/LoginActivity.kt
package nuricanozturk.dev.android.multipleactivity

import android.content.Intent
import android.os.Build
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import androidx.annotation.RequiresApi
import androidx.databinding.DataBindingUtil
import nuricanozturk.dev.android.multipleactivity.databinding.ActivityLoginBinding

import nuricanozturk.dev.android.multipleactivity.keys.PRODUCT_NAME

import nuricanozturk.dev.android.multipleactivity.keys.TOTAL_PRICE

import nuricanozturk.dev.android.multipleactivity.keys.USER_INFO
import nuricanozturk.dev.android.multipleactivity.viewmodel.LoginActivityViewModel
import nuricanozturk.dev.android.multipleactivity.viewmodel.LoginInfoViewModel
import java.time.LocalDateTime
import java.time.format.DateTimeFormatter

const val PAYMENT_ACTIVITY_REQUEST_CODE = 1

class LoginActivity : AppCompatActivity()
{

    lateinit var mBinding : ActivityLoginBinding

    private fun initialize()
    {
        mBinding = DataBindingUtil.setContentView(this, R.layout.activity_login)
        mBinding.viewModel = LoginActivityViewModel(this)
        mBinding.loginInfoViewModel = LoginInfoViewModel()
    }

    override fun onCreate(savedInstanceState : Bundle?)
    {
        super.onCreate(savedInstanceState)
        initialize()
    }

    @RequiresApi(Build.VERSION_CODES.O)
    fun handleLoginButton()
    {
        val now = LocalDateTime.now().format(DateTimeFormatter.ofPattern("dd/MM/yyyy hh:mm:ss"))
            .toString()

        Toast.makeText(this, now, Toast.LENGTH_LONG).show()

        Intent(this, PaymentActivity::class.java).apply {
            startActivityForResult(putExtra(USER_INFO, mBinding.loginInfoViewModel), PAYMENT_ACTIVITY_REQUEST_CODE)
        }
    }

    override fun onActivityResult(requestCode : Int, resultCode : Int, data : Intent?)
    {
        if (requestCode == PAYMENT_ACTIVITY_REQUEST_CODE && resultCode == RESULT_OK)
        {
            val productName = data?.getStringExtra(PRODUCT_NAME)
            val totalPrice = data?.getDoubleExtra(TOTAL_PRICE, 0.0)

            "$totalPrice paid for '$productName'".apply {
                Toast.makeText(this@LoginActivity, this, Toast.LENGTH_LONG).show()
            }

        }
        super.onActivityResult(requestCode, resultCode, data)
    }
}